src/Eloquent/Traits/HasAdjacencyList.php

```diff
@@ -203,27 +205,29 @@ protected function newBloodline(Builder $query, Model $parent, $foreignKey, $loc
     /**
      * Get the model's children.
      *
-     * @return \Illuminate\Database\Eloquent\Relations\HasMany<self>
+     * @return \Illuminate\Database\Eloquent\Relations\HasMany<$this>
      */
     public function children()
     {
+        // @phpstan-ignore return.type
         return $this->hasMany(static::class, $this->getParentKeyName(), $this->getLocalKeyName());
```

Comment:
```suggestion
     * @return \Illuminate\Database\Eloquent\Relations\HasMany<static>
     */
    public function children()
    {
            return $this->hasMany(static::class, $this->getParentKeyName(), $this->getLocalKeyName());
```

Reply:
Only works if the tests path is excluded